staging/src/k8s.io/apiserver/pkg/server/options/recommended.go

```diff
@@ -124,12 +125,16 @@ func (o *RecommendedOptions) ApplyTo(config *server.RecommendedConfig) error {
 		return err
 	}
 	if feature.DefaultFeatureGate.Enabled(features.APIPriorityAndFairness) {
-		config.FlowControl = utilflowcontrol.New(
-			config.SharedInformerFactory,
-			kubernetes.NewForConfigOrDie(config.ClientConfig).FlowcontrolV1beta1(),
-			config.MaxRequestsInFlight+config.MaxMutatingRequestsInFlight,
-			config.RequestTimeout/4,
-		)
+		if config.ClientConfig != nil {
+			config.FlowControl = utilflowcontrol.New(
+				config.SharedInformerFactory,
+				kubernetes.NewForConfigOrDie(config.ClientConfig).FlowcontrolV1beta1(),
+				config.MaxRequestsInFlight+config.MaxMutatingRequestsInFlight,
+				config.RequestTimeout/4,
+			)
+		} else {
+			klog.Warningf("neither kubeconfig is provided nor service-account is mounted, so APIPriorityAndFairness will be disabled")
```

Comment:
nit "Neither" since it's a message, not an error.

Reply:
fixed